compiler/src/iree/compiler/Dialect/VM/Conversion/ArithToVM/Patterns.cpp

```diff
@@ -539,6 +540,14 @@ struct IntToFPOpConversion : public OpConversionPattern<OpTy> {
       return rewriter.notifyMatchFailure(srcOp, "unsupported type");
     }
     Value input = srcOp.getIn();
+    auto resultType = this->getTypeConverter()->convertType(dstType);
+    if (llvm::isa<arith::SIToFPOp>(srcOp) &&
```

Comment:
Please add a conversion pattern here: https://github.com/iree-org/iree/blob/ec8d4e8d5104bdd07c66fb3a71d105e6c182a977/compiler/src/iree/compiler/Dialect/VM/Conversion/ArithToVM/Patterns.cpp#L753 (Templated style) Also, don't add a check for the src op to be `arith.sitofp`. It's already known.

Reply:
@pashu123 adding a template style pattern will, I think, require me to specialize the struct, to choose between `SI32F32` and `SI64F32`. I felt that the current approach was a simpler route to take, so that is what I did.

Therefore, I added a check that the op is `arith.sitofp` as the same function is called for `arith.uitofp`, and the new conversion does not apply to the latter.

Do you want me to take the specialization route? Or do you have an alternative in mind?

Reply:
Yes, please take the specialization route; this clearly shows what ops are supported. Also, if you could make the current function more generic to handle si / ui of any bit width to be converted into fp of any bit width (legal cases), that would make life easier for someone adding new test cases. It would just be an op into a templatized function. (e.g., we have all the mechanisms for `SI64F46` conversion, but the function isn't generic enough to handle such a case.

Reply:
agreed @pashu123!